test/it/uk/gov/hmrc/exports/migrations/changelogs/submission/AddAssociatedSubmissionToDeclarationsISpec.scala

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.exports.migrations.changelogs.submission
+
+import org.bson.Document
+import uk.gov.hmrc.exports.base.IntegrationTestMigrationToolSpec
+import uk.gov.hmrc.exports.models.declaration.DeclarationStatus._
+
+class AddAssociatedSubmissionIdToDeclarationsISpec extends IntegrationTestMigrationToolSpec {
+
+  import AddAssociatedSubmissionIdToDeclarationsISpec._
+
+  override val collectionUnderTest = "declarations"
+  override val changeLog = new AddAssociatedSubmissionIdToDeclarations()
+
+  "AddAssociatedSubmissionIdToDeclarations" should {
+
+    "Adds associatedSubmissionId when submission contains one action with related decId" in {
+      setupSubmissions(submissionWithSingleAction())
+      runTest(declarationBeforeMigration(), declarationAfterMigration())
+    }
+
+    "Adds associatedSubmissionId when submission contains multiple action with one related decId" in {
+      setupSubmissions(submissionWithMultipleActions)
+      runTest(declarationBeforeMigration(), declarationAfterMigration())
+    }
+
+    "Sets declaration's associatedSubmissionId value" when {
+      "declaration's status is AMENDMENT_DRAFT" in {
+        setupSubmissions(submissionWithMultipleActions)
+        runTest(declarationBeforeMigration(AMENDMENT_DRAFT), declarationAfterMigration(AMENDMENT_DRAFT))
+      }
+      "declaration's status is COMPLETE" in {
+        setupSubmissions(submissionWithMultipleActions)
+        runTest(declarationBeforeMigration(COMPLETE), declarationAfterMigration(COMPLETE))
+      }
+    }
+
+    "Does not set the declaration's associatedSubmissionId value" when {
+      "declaration's status is DRAFT" in {
+        setupSubmissions(submissionWithMultipleActions)
+        runTest(declarationBeforeMigration(DRAFT), declarationBeforeMigration(DRAFT))
+      }
+      "associatedSubmissionId already defined" in {
+        setupSubmissions(submissionWithSingleAction())
+        runTest(declarationAfterMigration(), declarationAfterMigration())
+      }
+    }
+
+    "Logs declaration not updated" when {
```

Comment:
Is some kind of log check occurring on these? Can't see anything?

Reply:
It was my initial intention to add some checks of the logging but after looking into how to do this with Play it was a non-trivial task to be able to verify log messages during test runs. So instead I manually checked the logs and left the tests in to check that the records were not updated if they ran into any issues